client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -181,7 +181,22 @@ class SimplePaymentsDialog extends Component {
 		return new Promise( resolve => this.formStateController.handleSubmit( resolve ) );
 	}
 
-	handleChangeTabs = activeTab => this.setState( { activeTab } );
+	handleChangeTabs = activeTab => {
+		if ( activeTab === 'form' ) {
+			this.editPaymentButton( null );
+		} else {
+			this.showButtonList();
+		}
+	};
+
+	showButtonList() {
+		this.setState( { activeTab: 'list' } );
+	}
+
+	editPaymentButton = editedPaymentId => {
```

Comment:
Nit: I would call this `showEditPaymentButton`. It's more clear that we are switching the views and also `showButtonList` uses the same convention.